sklearn/calibration.py

```diff
@@ -282,25 +289,25 @@ def fit(self, X, y, sample_weight=None, **fit_params):
         for sample_aligned_params in fit_params.values():
             check_consistent_length(y, sample_aligned_params)
 
-        if self.base_estimator is None:
+        if self.estimator is None:
```

Comment:
so this is here that you should check for `self.base_estimator` and `self.estimator`. I think that you should raise an error if both are set.